php-frontend/src/test/java/org/sonar/php/filters/SuppressWarningFilterTest.java

```diff
@@ -100,20 +103,190 @@ void filterOutIssueMultipleRule(String suppressWarning) throws URISyntaxExceptio
 
   @ParameterizedTest
   @ValueSource(strings = {
-    // "#[SuppressWarnings(\"php:S1234\")]", // TODO : fix
     "// @SuppressWarnings(\"php:S1234\")",
     "# @SuppressWarnings(\"php:S1234\")",
     "/* @SuppressWarnings(\"php:S1234\") */",
   })
   void filterOutIssueCommentSeparatedByEmptyLine(String suppressWarning) throws URISyntaxException {
-    PhpFile file = prepareFile("myFile.php");
-    String code = asCode("<?php",
-      suppressWarning,
-      "",
-      "function foo(){} ");
-    SuppressWarningFilter suppressWarningFilter = new SuppressWarningFilter();
-    suppressWarningFilter.analyze(file, parseSource(code));
-    assertThat(suppressWarningFilter.accept("myFile.php", "php:S1234", 4)).isFalse();
+    assertScopeOfSuppressWarningInstruction(
+      "",          "<?php",
+      "",          suppressWarning,
+      "",          "",
+      "php:S1234", "function foo(){} "
+    );
+  }
+
+  @Test
+  void filterOutIssueAttributeSeparatedByEmptyLine() throws URISyntaxException {
+    assertScopeOfSuppressWarningInstruction(
+      "",          "<?php",
+      "php:S1234", "#[SuppressWarnings(\"php:S1234\")]",
+      "php:S1234", "",
+      "php:S1234", "function foo(){} "
+    );
```

Comment:
while it is a bit of a hacky solution to use it like this (in all methods), it is really really readable!